### ISSUE TITLE
Introduce version in the trigger spec

### ISSUE THREAD

**Issue by tomkerkhove** [kedacore/keda#613]
Introduce version in the trigger spec so that we can release new versions of the trigger specification without breaking customers.

This allows us to introduce new ways of describing things without having messy documentation and is easier to troubleshoot.

### Use-Case

When a scaler decides to change things around by renaming fields, moving them around or introduce breaking changes there is no easy way to see the difference as a customer.

With clear versioning it would be more straight forward and the documentation can be built around it.

### Current

```yaml
triggers:
  - type: kafka
    metadata:
      brokerList: kafka.svc:9092
      consumerGroup: my-group
      topic: test-topic
      lagThreshold: '5'
```

### Proposal

We could add a `version` to the trigger spec:

```yaml
triggers:
  - type: kafka
    version: v1
    metadata:
      brokerList: kafka.svc:9092
      consumerGroup: my-group
      topic: test-topic
      lagThreshold: '5'
```

### Impact

The impact should be fairly minimal

- Configuration parsing should be able to coop with this
  - When no version is specified `v1` should be assumed
- Documentation should be updated
- Releases should note if new versions are released

**Comment by tomkerkhove** [comment]
We should also look at versioning scalers independently from the core, but that might be another issue.

**Comment by tomkerkhove** [comment]
Re-opening as I think it's time to do this.

**Comment by JorTurFer** [comment]
I think this could be a huge headache because we'd have potentially twice or 3 times more code to maintain. I'm not saying no to this, but before starting with this, I'd like to have the versioning lifecycle defined and what users can expect from each version. 
Are we going to apply backports with fixes or we only actively maintain latest versions? What about e2e tests? How long are we going to support old versions? Is our documentation page ready to offering this feature?
I see the potential of this feature, and it could be a nice improvement, but maybe we should define the surrounding things before. WDYT?

**Comment by tomkerkhove** [comment]
Oh yes, I'm not saying that we should already implement this straight away but I think we _have_ to do it. It's either this way, or we cannot make breaking changes to triggers unless we bump the API version for our _whole_ CRD.

Before I went in to detail, I wanted to see if we all agree this is something we should do. In terms of duplication, I'd say that this is just something we can isolate from the configuration parsing and share the logic (unless big things are being removed), but I think it's mainly for the way we configure things.

With regards to end-to-end tests, I'd say we support current + current-1, but that's just a proposal
